addons/mail/static/src/core/common/thread_service.js

```diff
@@ -480,7 +480,19 @@ export class ThreadService {
         }
     }
 
-    async getChat({ userId, partnerId }) {
+    /**
+     * Search for a chat relaetd info with a given user or partner id.
+     * It returns the chat thread if it exists,
+     * otherwise it returns the user AND partner ids based on the given parameters.
+     * @param {Object} param0
+     * @param {number} param0.userId
+     * @param {number} param0.partnerId
+     * @returns {Object} result
+     * @returns {number} result.userId
+     * @returns {number} result.partnerId
+     * @returns {Thread} result.chat
+     */
+    async searchChat({ userId, partnerId }) {
```

Comment:
This function is too complex, and it shows by the return values that should just be `chat`.
The reason you want to return `userId` and `partnerId` is because this function fetches `userId` and `partnerId`, in order to determine some conditions for chats.

It's better to isolate that part to a dedicated function, e.g. `getUserAndPartner({ userId, partnerId })`, which returns the persona and internally fetches missing data.

That way, you have a function for just fetching data for user/partner on the persona, and then you can a function for `searchChat`.

===

Note that it's weird to have `getChat` and `searchChat`. Current `getChat` is poorly named: It should be removed and call `startChat` instead.
Code of `getChat` sucks. Better keep that way in your PR for the moment, we'll do some code cleaning soon (not in this PR)

Reply:
I find splitting `getChat` and `searchChat` kind of makes sense. Because in my case, I don't want to join the chat but just want to know if the chat is exist or not. By using `getChat`/`startChat` you cannot do that.

Reply:
I understand with current `getChat` that is poorly named and should be instead `startChat`. If we had this code cleaned (not to do in this PR), then we can remove current `getChat` and reserve this name for the `searchChat` that you just introduced.